LabApiUtilities/src/test/com/microsoft/identity/labapi/utilties/authentication/common/CertificateCredentialTest.java

```diff
@@ -0,0 +1,252 @@
+                    }
+
+                    @Override
+                    public void verify(PublicKey key) throws CertificateException, NoSuchAlgorithmException, InvalidKeyException, NoSuchProviderException, SignatureException {
+
+                    }
+
+                    @Override
+                    public void verify(PublicKey key, String sigProvider) throws CertificateException, NoSuchAlgorithmException, InvalidKeyException, NoSuchProviderException, SignatureException {
+
+                    }
+
+                    @Override
+                    public String toString() {
+                        return null;
+                    }
+
+                    @Override
+                    public PublicKey getPublicKey() {
+                        return null;
+                    }
+
+                    @Override
+                    public boolean hasUnsupportedCriticalExtension() {
+                        return false;
+                    }
+
+                    @Override
+                    public Set<String> getCriticalExtensionOIDs() {
+                        return null;
+                    }
+
+                    @Override
+                    public Set<String> getNonCriticalExtensionOIDs() {
+                        return null;
+                    }
+
+                    @Override
+                    public byte[] getExtensionValue(String oid) {
+                        return new byte[0];
+                    }
+                }
+        );
+
+        Assert.assertNotNull(certificateCredential);
+        Assert.assertNotNull(certificateCredential.getPrivateKey());
+        Assert.assertNotNull(certificateCredential.getPublicCertificate());
+    }
+
+    @Test
+    public void testCanCertificateCredentialFromKeyStoreConfigurationAndCertificateMetadata() {
+        final CertificateCredential certificateCredential;
+        try {
+            certificateCredential = CertificateCredential.create(
+                    new KeyStoreConfiguration(
+                            KEYSTORE_TYPE, KEYSTORE_PROVIDER, null
+                    ),
+                    new ClientCertificateMetadata(CERTIFICATE_ALIAS_VALID, null)
+            );
+        } catch (LabApiException e) {
+            throw new AssertionError(e);
+        }
+
+        Assert.assertNotNull(certificateCredential);
+        Assert.assertNotNull(certificateCredential.getPrivateKey());
+        Assert.assertNotNull(certificateCredential.getPublicCertificate());
+    }
+
+    @Test
+    public void testCannotCreateCertificateCredentialIfCertNotFoundInKeyStore() {
+        try {
+            CertificateCredential.create(
+                    new KeyStoreConfiguration(
+                            KEYSTORE_TYPE, KEYSTORE_PROVIDER, null
+                    ),
+                    new ClientCertificateMetadata(CERTIFICATE_ALIAS_INVALID, null)
+            );
+            Assert.fail("We weren't expecting to hit this line...exception should've already occurred");
```

Comment:
throw new AssertionError...?

Reply:
I usually do that when I have a handle on another throwable...and just wrap it into `AssertionError`..in this case I don't have one so the effect of throw new AssertionError and Assert.fail..should be the same